neqo-transport/src/connection/tests/mod.rs

```diff
@@ -247,51 +289,26 @@ fn exchange_ticket(
     get_tokens(client).pop().expect("should have token")
 }
 
-/// Getting the client and server to reach an idle state is surprisingly hard.
-/// The server sends `HANDSHAKE_DONE` at the end of the handshake, and the client
-/// doesn't immediately acknowledge it.  Reordering packets does the trick.
-fn force_idle(
-    client: &mut Connection,
-    server: &mut Connection,
-    rtt: Duration,
-    mut now: Instant,
-) -> Instant {
-    // The client has sent NEW_CONNECTION_ID, so ensure that the server generates
-    // an acknowledgment by sending some reordered packets.
-    qtrace!("force_idle: send reordered client packets");
-    let c1 = send_something(client, now);
-    let c2 = send_something(client, now);
-    now += rtt / 2;
-    server.process_input(&c2, now);
-    server.process_input(&c1, now);
-
-    // Now do the same for the server.  (The ACK is in the first one.)
-    qtrace!("force_idle: send reordered server packets");
-    let s1 = send_something(server, now);
-    let s2 = send_something(server, now);
-    now += rtt / 2;
-    // Delivering s2 first at the client causes it to want to ACK.
-    client.process_input(&s2, now);
-    // Delivering s1 should not have the client change its mind about the ACK.
-    let ack = client.process(Some(&s1), now);
-    assert!(ack.as_dgram_ref().is_some());
+/// The `handshake` method inserts PING frames into the first application data packets,
+/// which forces each peer to ACK them. As a side effect, that causes both sides of the
+/// connection to be idle aftwerwards. This method simply verifies that this is the case.
+fn assert_idle(client: &mut Connection, server: &mut Connection, rtt: Duration, now: Instant) {
     let idle_timeout = min(
         client.conn_params.get_idle_timeout(),
         server.conn_params.get_idle_timeout(),
     );
-    assert_eq!(client.process_output(now), Output::Callback(idle_timeout));
-    now += rtt / 2;
+    // Client started its idle period half an RTT before now.
     assert_eq!(
-        server.process(ack.as_dgram_ref(), now),
+        client.process_output(now.checked_sub(rtt / 2).unwrap()),
         Output::Callback(idle_timeout)
     );
```

Comment:
I think that you should subtract the half RTT from the expectation, rather than calling process_output in (what might be) the past.

There is a potential problem with rounding errors (if the RTT is an odd number of nanoseconds), but I think we're OK in that regard, because the test rounds the RTT down when it increments the time.